File: app/src/main/java/io/github/freewebmovement/igniter/activities/DomainMonitorActivity.java
package io.github.freewebmovement.igniter.activities;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.FragmentManager;

import io.github.freewebmovement.igniter.R;

/**
 * Thin wrapper keeping the standalone entry point alive; the actual UI lives
 * in {@link RulesFragment} which is also used by the main tab shell.
 */
public class DomainMonitorActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_domain_monitor);
        FragmentManager fm = getSupportFragmentManager();
        RulesFragment fragment = (RulesFragment) fm.findFragmentByTag("rules");
        if (fragment == null) {
            fragment = RulesFragment.newInstance();
        }
        fm.beginTransaction()
                .replace(R.id.parent_fl, fragment, "rules")
                .commitAllowingStateLoss();
    }
}

File: app/src/main/java/io/github/freewebmovement/igniter/activities/DomainMonitorActivity.kt
package io.github.freewebmovement.igniter.activities

import android.os.Bundle
import android.view.View
import android.widget.FrameLayout
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.FragmentManager

/**
 * Thin wrapper keeping the standalone entry point alive; the actual UI lives
 * in [RulesFragment] which is also used by the main tab shell.
 */
class DomainMonitorActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val container = FrameLayout(this)
        container.id = View.generateViewId()
        setContentView(container)
        val fm: FragmentManager = supportFragmentManager
        val fragment = fm.findFragmentByTag("rules") as? RulesFragment
            ?: RulesFragment.newInstance()
        fm.beginTransaction()
            .replace(container.id, fragment, "rules")
            .commitAllowingStateLoss()
    }
}
